File: backoffice/src/table/domain/TableRepository.py
"""
 *
 * Libraries 
 *
"""

from abc               import abstractmethod
from .Table            import Table
from .TableNumber      import TableNumber
from src.shared.domain import TableId
from src.shared.domain import RestaurantId


"""
 *
 * Interfaces
 *
"""

class TableRepository:

    """
     *
     * Methods 
     *
    """
    
    @abstractmethod
    def insert( self, table : Table ) -> bool:
        pass
    
    @abstractmethod
    def update( self, table : Table ) -> bool:
        pass
    
    @abstractmethod
    def selectByIdAndRestaurant( self, id : TableId, restaurantId : RestaurantId ) -> Table:
        pass
    
    @abstractmethod
    def selectByNumberAndRestaurant( self, number : TableNumber, restaurantId : RestaurantId ) -> Table:
        pass
    
    @abstractmethod
    def selectAllByRestaurant( self, restaurantId : RestaurantId ) -> list:
        pass

File: attention/src/table/domain/TableRepository.cs
/* 
 * 
 * Libraries
 *
*/

using attention.src.shared.domain;

/* 
 * 
 * Interfaces
 *
*/

namespace attention.src.table.domain {

    public interface TableRepository {

        /* 
         * 
         * Methods
         *
        */

        bool insert( Table table );
        bool update( Table table );
        Table? findById( TableId id );


    }

}